aem-authoring-toolkit-api/src/main/java/com/exadel/aem/toolkit/api/annotations/widgets/DataSource.java

```diff
@@ -0,0 +1,30 @@
+package com.exadel.aem.toolkit.api.annotations.widgets;
+
+import com.exadel.aem.toolkit.api.annotations.meta.PropertyMapping;
+import com.exadel.aem.toolkit.api.annotations.widgets.property.Property;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target(ElementType.FIELD)
```

Comment:
Common class-wide javadoc is also reqired))
Like 
```
/**
* Allows to define a {@code datasource} node for a {@link @Select or a @RadioGroup} annotation as described in <a href="...">Adobe Documentation</a>. Use this to specify either an ACS Commons list source or a custom datasource for your selection,
*/
```

UPD: I looked beneath and saw you explain the datasource meaning per-field. But anyway... let it be here also